src/utils/mod.rs

```diff
@@ -27,14 +28,19 @@ pub(crate) mod notify;
 pub mod raw;
 pub(crate) mod units;
 
-pub fn io_thread_count() -> usize {
-    match thread::available_parallelism() {
+pub fn io_thread_count(process: &Process) -> anyhow::Result<usize> {
+    if let Ok(n) = process.var("RUSTUP_IO_THREADS") {
+        return usize::from_str(&n)
+            .context("invalid value in RUSTUP_IO_THREADS. Must be a natural number");
```

Comment:
I'm afraid that has to be a *positive* number (`NonZero<usize>`), or you have something else planned for `0`?